From e8b3eb4f9715f2a73dca9ddfc612da523a40436c Mon Sep 17 00:00:00 2001
From: Jianzheng Zhou <jianzheng.zhou@freescale.com>
Date: Mon, 2 Sep 2013 13:40:08 +0800
Subject: [PATCH 10/11] ENGR00256988 3G:delete redundant set on route

Delete setting route here which violates "Must NOT change the linux
routing table." rule described in ril.h. Route should be gotten in
response of SETUP DATA CALL.

Signed-off-by: Jianzheng Zhou <jianzheng.zhou@freescale.com>
---
 android/ip-up-ppp0.c | 40 +---------------------------------------
 1 file changed, 1 insertion(+), 39 deletions(-)

diff --git a/android/ip-up-ppp0.c b/android/ip-up-ppp0.c
index c88fd53..cd0b13c 100644
--- a/android/ip-up-ppp0.c
+++ b/android/ip-up-ppp0.c
@@ -1,5 +1,6 @@
 /*
  * Copyright (C) 2009 The Android Open Source Project
+ * Copyright (C) 2013 Freescale Semiconductor, Inc.
  *
  * Licensed under the Apache License, Version 2.0 (the "License");
  * you may not use this file except in compliance with the License.
@@ -38,9 +39,6 @@ static inline struct in_addr *in_addr(struct sockaddr *sa)
 
 int main(int argc, char **argv)
 {
-    struct rtentry rt;
-    int s, ret;
-
 /* args is like this:
     argv[1] = ifname;
     argv[2] = devnam;
@@ -59,42 +57,6 @@ int main(int argc, char **argv)
     __android_log_print(ANDROID_LOG_INFO, "ip-up-ppp0", "\tLocal IP\t:\t%s",	argv[4]);
     __android_log_print(ANDROID_LOG_INFO, "ip-up-ppp0", "\tRemote IP\t:\t%s",	argv[5]);
 
-    /* Clean out the RTREQ structure */
-    memset(&rt, 0, sizeof(rt));
-
-    /* Prepare new "default" route entry for ppp0 link 
-       Destination: 0.0.0.0
-       Netmask:     0.0.0.0
-       Gateway:     not specified 
-       Interface:   ifname as argv[1] 
-       Flag:	    RTF_UP
-    */
-    rt.rt_dst.sa_family = AF_INET;
-    ((struct sockaddr_in *)&rt.rt_dst)->sin_addr.s_addr = INADDR_ANY;
-
-    rt.rt_genmask.sa_family = AF_INET;
-    ((struct sockaddr_in *)&rt.rt_genmask)->sin_addr.s_addr = INADDR_ANY;
-
-    rt.rt_dev = argv[1];
-
-    rt.rt_flags = RTF_UP;
-    /* Add "default" entry into route table */
-    s = socket(AF_INET, SOCK_DGRAM, 0);
-    if(s == -1) {
-        __android_log_print(ANDROID_LOG_ERROR, "ip-up-ppp0", "Fail to get socket to modify route table");
-	return EPERM;
-    } 	
-    ret = ioctl(s, SIOCADDRT, &rt);
-    if(ret == -1) {
-	if(errno == EEXIST) {
-            __android_log_print(ANDROID_LOG_WARN, "ip-up-ppp0", "Unable to add default entry in route table - Already exist");
-	    return 0;
-	}
-        __android_log_print(ANDROID_LOG_ERROR, "ip-up-ppp0", "Fail to add default entry in route table - %s", strerror(errno));
-	return EPERM;
-    }
-    __android_log_print(ANDROID_LOG_INFO, "ip-up-ppp0", "Add default entry into route table");
-
     {
 	// we need setup lots of property like net.$LINKNAME.[local-ip|remote-ip|dns1|dns2]
 	strcpy(propstr, "net."); 
-- 
1.8.0

